### DISCUSSION TITLE
postgres

### DISCUSSION THREAD

**Discussion by jllipatz** [duckdb/duckdb#4980] [Q&A]
Hello,

I don't understand how to use duckdb extensions.

The INSTALL command doesn't work and I downloaded by hand the extensions files in my directory .duckb/extensions/v0.5.1/windows_amd64. I installed fts which is documented on duckdb.org and postgres_scanner which is not. 
But when I run the LOAD command on either of the two extensions, it terminates my R session without any message while a LOAD of another name simply exits on error. It seems to indicate that I put things at the right place. What is wrong?

I saw a similar question with Python and I am also working within Windows10.

Best regards

**Comment by Alex-Monahan** [comment]
Sorry to hear that!
Could you provide the exact code you are running? That may help us narrow things down. The extensions are also version specific, so are you using the same version for DuckDB and the extensions?

**Reply by jllipatz** [reply]
Really sorry too!

The code is quite simple : 
`library(DBI)
con <- dbConnect(duckdb::duckdb())

dbSendQuery(con, "LOAD postgres_scanner")
`
I run the version 0.5.1 of duckdb and tied R 4.2.1 and R 4.1.3 (with the same effect).

The message when trying the INSTALL command gave me something like that
`Erreur : rapi_execute: Failed to run query
Error: IO Error: Failed to download extension "json" at URL "http://extensions.duckdb.org/v0.5.1/windows_amd64/json.duckdb_extension.gz"
Extension "json" is an existing extension.
Are you using a development build? In this case, extensions might not (yet) be uploaded.`
It gave me the name of the file I downloaded, unzipped and put in a place that seems correct, since now `INSTALL postgres`_scanner returns immediatly.

**Reply by Mause** [reply]
> Really sorry too!
> 
> The code is quite simple : 
> `library(DBI)
> con <- dbConnect(duckdb::duckdb())
> 
> dbSendQuery(con, "LOAD postgres_scanner")
> `
> I run the version 0.5.1 of duckdb and tied R 4.2.1 and R 4.1.3 (with the same effect).
> 
> The message when trying the INSTALL command gave me something like that
> `Erreur : rapi_execute: Failed to run query
> Error: IO Error: Failed to download extension "json" at URL "http://extensions.duckdb.org/v0.5.1/windows_amd64/json.duckdb_extension.gz"
> Extension "json" is an existing extension.
> Are you using a development build? In this case, extensions might not (yet) be uploaded.`
> It gave me the name of the file I downloaded, unzipped and put in a place that seems correct, since now `INSTALL postgres`_scanner returns immediatly.

That error is strange, I'm able to download from that URL with no issues - are you behind a proxy, or working with an unstable internet connection?

For your manual attempt, are you decompressing the file first?

**Reply by jllipatz** [reply]
Yes I am behind a proxy, It is current that we cannot download directly files.
And of course I decompressed the file. Both zipped and unzipped are actually in the same directory.

**Reply by jllipatz** [reply]
I managed to install and load the extension on a linux machine. But unfortunately, from there, I have no access to the databases I want to use.
The problem I have encountered sounds like an extension not compatible with my windows machine. Is there some requirement on the version of the operating system?

**Reply by Mause** [reply]
To clarify, you're able to load the extensions successfully from R on a Linux machine? 

If possible, would you be able to provide the R core dump file for the windows machine you're using? That might help us track down the issue

**Reply by jllipatz** [reply]
From a fresh crash, in the windows log:
`Version=1
EventType=BEX64
EventTime=133124676550357666
ReportType=2
Consent=1
UploadTime=133124676568460111
ReportStatus=268435456
ReportIdentifier=44fc4d13-2f1e-44ca-9521-30d3bf77d3c2
IntegratorReportIdentifier=f1217c8a-2447-484c-9389-ce27719d87bd
Wow64Host=34404
NsAppName=Rgui.exe
AppSessionGuid=00003a38-0001-0041-1967-bbeb2ff4d801
TargetAppId=W:0006897e380d0e92bb4aab9c14cbc1a65fb600000904!0000b6fdc825452b7a43ae3e9241750af7b9ac43475b!Rgui.exe
TargetAppVer=2022//06//23:09:23:49!20765!Rgui.exe
BootId=4294967295
TargetAsId=4372
UserImpactVector=808452912
IsFatal=1
EtwNonCollectReason=1
Response.BucketId=53feafd66704c14b060cd35d28c4f441
Response.BucketTable=5
Response.LegacyBucketId=1588877165624357953
Response.type=4
Sig[0].Name=Nom de l’application
Sig[0].Value=Rgui.exe
Sig[1].Name=Version de l’application
Sig[1].Value=4.21.16977.0
Sig[2].Name=Horodatage de l’application
Sig[2].Value=62b43125
Sig[3].Name=Nom du module par défaut
Sig[3].Value=StackHash_7884
Sig[4].Name=Version du module par défaut
Sig[4].Value=0.0.0.0
Sig[5].Name=Horodateur du module par défaut
Sig[5].Value=00000000
Sig[6].Name=Décalage de l’exception
Sig[6].Value=PCH_1A_FROM_ntdll+0x000000000009DC14
Sig[7].Name=Code de l’exception
Sig[7].Value=c0000005
Sig[8].Name=Données d’exception
Sig[8].Value=0000000000000008
DynamicSig[1].Name=Version du système
DynamicSig[1].Value=10.0.19044.2.0.0.256.4
DynamicSig[2].Name=Identificateur de paramètres régionaux
DynamicSig[2].Value=1036
DynamicSig[22].Name=Information supplémentaire n° 1
DynamicSig[22].Value=7884
DynamicSig[23].Name=Information supplémentaire n° 2
DynamicSig[23].Value=7884c52f297098486eeeeebbd7f92329
DynamicSig[24].Name=Information supplémentaire n° 3
DynamicSig[24].Value=d081
DynamicSig[25].Name=Information supplémentaire n° 4
DynamicSig[25].Value=d081fb1e439a3bd45206cb8e47b8f5bd
UI[2]=C:\Users\h2izgk\AppData\Local\Programs\R\R-4.2.1\bin\x64\Rgui.exe
LoadedModule[0]=C:\Users\h2izgk\AppData\Local\Programs\R\R-4.2.1\bin\x64\Rgui.exe
LoadedModule[1]=C:\WINDOWS\SYSTEM32\ntdll.dll
LoadedModule[2]=C:\WINDOWS\System32\KERNEL32.DLL
LoadedModule[3]=C:\WINDOWS\System32\KERNELBASE.dll
LoadedModule[4]=C:\WINDOWS\System32\ucrtbase.dll
LoadedModule[5]=C:\WINDOWS\System32\USER32.dll
LoadedModule[6]=C:\WINDOWS\System32\win32u.dll
LoadedModule[7]=C:\WINDOWS\System32\GDI32.dll
LoadedModule[8]=C:\WINDOWS\System32\gdi32full.dll
LoadedModule[9]=C:\WINDOWS\System32\msvcp_win.dll
LoadedModule[10]=C:\Users\h2izgk\AppData\Local\Programs\R\R-4.2.1\bin\x64\Rgraphapp.dll
LoadedModule[11]=C:\WINDOWS\System32\comdlg32.dll
LoadedModule[12]=C:\Users\h2izgk\AppData\Local\Programs\R\R-4.2.1\bin\x64\R.dll
LoadedModule[13]=C:\WINDOWS\System32\msvcrt.dll
LoadedModule[14]=C:\WINDOWS\System32\ADVAPI32.dll
LoadedModule[15]=C:\WINDOWS\System32\combase.dll
LoadedModule[16]=C:\WINDOWS\System32\RPCRT4.dll
LoadedModule[17]=C:\WINDOWS\System32\sechost.dll
LoadedModule[18]=C:\WINDOWS\System32\shcore.dll
LoadedModule[19]=C:\WINDOWS\System32\SHLWAPI.dll
LoadedModule[20]=C:\WINDOWS\System32\SHELL32.dll
LoadedModule[21]=C:\WINDOWS\System32\IMM32.dll
LoadedModule[22]=C:\WINDOWS\WinSxS\amd64_microsoft.windows.common-controls_6595b64144ccf1df_6.0.19041.1110_none_60b5254171f9507e\COMCTL32.dll
LoadedModule[23]=C:\Users\h2izgk\AppData\Local\Programs\R\R-4.2.1\bin\x64\Rblas.dll
LoadedModule[24]=C:\Users\h2izgk\AppData\Local\Programs\R\R-4.2.1\bin\x64\Riconv.dll
LoadedModule[25]=C:\WINDOWS\SYSTEM32\VERSION.dll
LoadedModule[26]=C:\WINDOWS\SYSTEM32\MSIMG32.dll
LoadedModule[27]=C:\WINDOWS\SYSTEM32\WINMM.dll
LoadedModule[28]=C:\WINDOWS\system32\uxtheme.dll
LoadedModule[29]=C:\WINDOWS\System32\ole32.dll
LoadedModule[30]=C:\WINDOWS\SYSTEM32\windows.storage.dll
LoadedModule[31]=C:\WINDOWS\SYSTEM32\Wldp.dll
LoadedModule[32]=C:\WINDOWS\System32\OLEAUT32.dll
LoadedModule[33]=C:\WINDOWS\SYSTEM32\PROPSYS.dll
LoadedModule[34]=C:\WINDOWS\SYSTEM32\profapi.dll
LoadedModule[35]=C:\WINDOWS\SYSTEM32\kernel.appcore.dll
LoadedModule[36]=C:\WINDOWS\System32\bcryptPrimitives.dll
LoadedModule[37]=C:\WINDOWS\System32\clbcatq.dll
LoadedModule[38]=C:\WINDOWS\System32\MSCTF.dll
LoadedModule[39]=C:\WINDOWS\SYSTEM32\textinputframework.dll
LoadedModule[40]=C:\WINDOWS\SYSTEM32\CoreUIComponents.dll
LoadedModule[41]=C:\WINDOWS\SYSTEM32\CoreMessaging.dll
LoadedModule[42]=C:\WINDOWS\System32\WS2_32.dll
LoadedModule[43]=C:\WINDOWS\SYSTEM32\wintypes.dll
LoadedModule[44]=C:\WINDOWS\SYSTEM32\ntmarta.dll
LoadedModule[45]=C:\WINDOWS\SYSTEM32\TextShaping.dll
LoadedModule[46]=C:\WINDOWS\SYSTEM32\SspiCli.dll
LoadedModule[47]=C:\Users\h2izgk\AppData\Local\Programs\R\R-4.2.1\library\methods\libs\x64\methods.dll
LoadedModule[48]=C:\Users\h2izgk\AppData\Local\Programs\R\R-4.2.1\library\utils\libs\x64\utils.dll
LoadedModule[49]=C:\Users\h2izgk\AppData\Local\Programs\R\R-4.2.1\library\grDevices\libs\x64\grDevices.dll
LoadedModule[50]=C:\Users\h2izgk\AppData\Local\Programs\R\R-4.2.1\library\graphics\libs\x64\graphics.dll
LoadedModule[51]=C:\Users\h2izgk\AppData\Local\Programs\R\R-4.2.1\library\stats\libs\x64\stats.dll
LoadedModule[52]=C:\Users\h2izgk\AppData\Local\Programs\R\R-4.2.1\bin\x64\Rlapack.dll
LoadedModule[53]=C:\WINDOWS\SYSTEM32\riched20.dll
LoadedModule[54]=C:\WINDOWS\SYSTEM32\USP10.dll
LoadedModule[55]=C:\WINDOWS\SYSTEM32\msls31.dll
LoadedModule[56]=C:\WINDOWS\system32\dataexchange.dll
LoadedModule[57]=C:\WINDOWS\system32\dcomp.dll
LoadedModule[58]=C:\WINDOWS\system32\d3d11.dll
LoadedModule[59]=C:\WINDOWS\system32\dxgi.dll
LoadedModule[60]=C:\WINDOWS\system32\twinapi.appcore.dll
LoadedModule[61]=C:\WINDOWS\SYSTEM32\edputil.dll
LoadedModule[62]=C:\Users\h2izgk\AppData\Local\Programs\R\R-4.2.1\library\duckdb\libs\x64\duckdb.dll
LoadedModule[63]=C:\WINDOWS\SYSTEM32\CRYPTBASE.DLL
LoadedModule[64]=C:\Users\h2izgk\.duckdb\extensions\v0.5.1\windows_amd64\postgres_scanner.duckdb_extension
LoadedModule[65]=C:\WINDOWS\System32\WLDAP32.dll
LoadedModule[66]=C:\WINDOWS\SYSTEM32\Secur32.dll
LoadedModule[67]=C:\WINDOWS\SYSTEM32\MSVCP140.dll
LoadedModule[68]=C:\WINDOWS\SYSTEM32\VCRUNTIME140.dll
LoadedModule[69]=C:\WINDOWS\SYSTEM32\WSOCK32.dll
LoadedModule[70]=C:\WINDOWS\SYSTEM32\VCRUNTIME140_1.dll
State[0].Key=Transport.DoneStage1
State[0].Value=1
OsInfo[0].Key=vermaj
OsInfo[0].Value=10
OsInfo[1].Key=vermin
OsInfo[1].Value=0
OsInfo[2].Key=verbld
OsInfo[2].Value=19044
OsInfo[3].Key=ubr
OsInfo[3].Value=2130
OsInfo[4].Key=versp
OsInfo[4].Value=0
OsInfo[5].Key=arch
OsInfo[5].Value=9
OsInfo[6].Key=lcid
OsInfo[6].Value=1036
OsInfo[7].Key=geoid
OsInfo[7].Value=84
OsInfo[8].Key=sku
OsInfo[8].Value=4
OsInfo[9].Key=domain
OsInfo[9].Value=1
OsInfo[10].Key=prodsuite
OsInfo[10].Value=256
OsInfo[11].Key=ntprodtype
OsInfo[11].Value=1
OsInfo[12].Key=platid
OsInfo[12].Value=10
OsInfo[13].Key=sr
OsInfo[13].Value=0
OsInfo[14].Key=tmsi
OsInfo[14].Value=577077
OsInfo[15].Key=osinsty
OsInfo[15].Value=3
OsInfo[16].Key=iever
OsInfo[16].Value=11.789.19041.0-11.0.1000
OsInfo[17].Key=portos
OsInfo[17].Value=0
OsInfo[18].Key=ram
OsInfo[18].Value=8067
OsInfo[19].Key=svolsz
OsInfo[19].Value=234
OsInfo[20].Key=wimbt
OsInfo[20].Value=0
OsInfo[21].Key=blddt
OsInfo[21].Value=191206
OsInfo[22].Key=bldtm
OsInfo[22].Value=1406
OsInfo[23].Key=bldbrch
OsInfo[23].Value=vb_release
OsInfo[24].Key=bldchk
OsInfo[24].Value=0
OsInfo[25].Key=wpvermaj
OsInfo[25].Value=0
OsInfo[26].Key=wpvermin
OsInfo[26].Value=0
OsInfo[27].Key=wpbuildmaj
OsInfo[27].Value=0
OsInfo[28].Key=wpbuildmin
OsInfo[28].Value=0
OsInfo[29].Key=osver
OsInfo[29].Value=10.0.19041.2130.amd64fre.vb_release.191206-1406
OsInfo[30].Key=buildflightid
OsInfo[30].Value={AA05BC7E-8C37-4095-9B4E-E65D4910D2B3}.201
OsInfo[31].Key=edition
OsInfo[31].Value=Enterprise
OsInfo[32].Key=ring
OsInfo[32].Value=Retail
OsInfo[33].Key=expid
OsInfo[34].Key=fconid
OsInfo[35].Key=containerid
OsInfo[36].Key=containertype
OsInfo[37].Key=edu
OsInfo[37].Value=0
FriendlyEventName=Fonctionnement arrêté
ConsentKey=BEX64
AppName=R for Windows GUI front-end
AppPath=C:\Users\h2izgk\AppData\Local\Programs\R\R-4.2.1\bin\x64\Rgui.exe
NsPartner=windows
NsGroup=windows8
ApplicationIdentity=B22E3E84E33B9D8AE9D5415C04A2106A
MetadataHash=-673098346`

**Reply by jllipatz** [reply]
Today I am able:

-    to install and load the postgres_scanner extension using RStudio on a Debian UNIX.
-    to load the extension from dbeaver on a Windows Server 2019 and to access correctly a Postgres database. The install doesn't work but we don't have access to internet  on these machines so I did it by hand in the suggested directory.

And I get crashes from RGui:

-   on my personal computer (Windows 10), manual install too
-   in sessions on Window Server 2019

I noticed an open bug in the github repository for postgres_scanner, perhaps should it be advisable to merge this discussion and the open bug (which has no positive answer yet).
